### ISSUE TITLE
Rendering errors on long pages

### ISSUE THREAD

**Issue by mehaase** [scrapinghub/splash#150]
I have found that very long pages have some PNG rendering bugs. For example, [here is a forum thread with infinite scroll](http://what.thedailywtf.com/t/papers-please/6851). I have written a Lua directive that scrolls the page to the bottom several times (to trigger infinite scroll) and then scrolls back to the top and takes a snapshot.

Here's what the QWebView looks like in a real window:

![discourse1](https://cloud.githubusercontent.com/assets/320904/5670674/02176e50-9750-11e4-8c21-0e193ca82d5f.png)

And here's what it looks like when rendered as a PNG (warning, this is _very_ tall):

![discourse2](https://cloud.githubusercontent.com/assets/320904/5670684/11a4b224-9750-11e4-8849-654b09a84ea9.png)

Notice that there's a huge white gap in the PNG that does not exist in the QWebView.

Here's an excerpt from my Lua script:

```lua
  function main(splash)
    local url = splash.args.url
    local page_count = splash.args.page_count or 3

    splash:autoload(splash.args.js_source)
    assert(splash:go(url))

    for i=1,page_count do
      scroll_to_bottom_and_wait(splash)
    end

    splash:runjs([[
      window.__scrollTop();
    ]])
    splash:stop()
    splash:set_viewport("full")

    return {
      html = splash:html(),
      png = splash:png{width=640},
      har = splash:har(),
    }
  end
```

Obviously I'm omitting quite a bit but I think this shows the core of the problem. This may not be a bug in splash -- I'm not sure. With even larger/long PNGs, the rendering is even further degraded, until it looks almost nothing like the original web page at all.

**Comment by kmike** [comment]
There are two known issues with these large pages.

The first issue is a QT bug which prevents a page from being rendered if one of the dimensions is greater than 32K. It is worked on here: https://github.com/scrapinghub/splash/pull/147.

The second issue is that viewport doesn't match window geometry, and it may confuse a webpage. It is worked on here: https://github.com/scrapinghub/splash/pull/131.

I'm not sure if these PRs will fix the problem described in this ticket or not, but they could. Similar issues:  https://github.com/scrapinghub/splash/issues/145, https://github.com/scrapinghub/splash/issues/80.

**Comment by kmike** [comment]
Could you try adding `self.web_view.setGeometry(0, 0, 1280, 1024)` here: https://github.com/scrapinghub/splash/blob/master/splash/browser_tab.py#L79 ? It is essentially a fix from #131. For me it fixed a similar issue on another website.

**Comment by mehaase** [comment]
Yes, that works! Thank you.
